sycl/include/syclcompat/util.hpp

```diff
@@ -42,6 +42,20 @@
 namespace syclcompat {
 
 namespace detail {
+
+template <typename tag, typename T> class generic_error_type {
+public:
+  generic_error_type() = default;
+  generic_error_type(T value) : value{value} {}
+  operator T() const { return value; }
+
+private:
+  T value;
+};
```

Comment:
This class does not do anything, outside of store a value.
I also do not see it used anywhere else, why are we adding this to the codebase?

Reply:
It's used by SYCLomatic to return a `generic_error_type` instead of a number for some error handling, just to mark what would be an int with some semantics. It's used with the err0 and err1 aliases. These are in detail as well for this snapshot, since it is confirmed they use it, I will move it those outside of the detail namespace